---
title: "creating-a-release"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{creating-a-release}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  eval=FALSE
)
```

# Creating a Release

This should come after the previous QC steps 
(batch effects, sex and expression). The metadata in the `dds` objects should 
be updated now with the current state of the database, meaning any additional 
suspicious sex samples or WGS identified mislabels, for instance, 
have been flagged in the database itself, and that updated information 
added back to the `dds` object.

## First, read in the data you wish to process (gene or tx)

```{r}
library(tidyverse)
library(DESeq2)
library(edgeR)
library(here)

# set environmental variables
## set to false to use gene level quantification, true to use
## tx level quantification
TX_QUANTS = FALSE
## determines whether data is written to the project data directory
WRITE = FALSE

# read in data ----------------------------------------------------------------
dds_list = list(
  gene = here(here("llfs_rnaseq_data/dds_20230713_gene.rds")),
  tx   = here(here("llfs_rnaseq_data/dds_20230713_tx.rds"))
)

# note, clear out any design -- it isn't necessary for the manipulations below.
# if you want to set the design, then change the formula below. Make sure to
# cast any character fields to factors.
if(TX_QUANTS){
  dds_full = readRDS(dds_list$tx)
  design(dds_full) = ~1
} else {
  dds_full = readRDS(dds_list$gene)
  design(dds_full) = ~1
}
```

## Next, split the dds object into `control` and `experiment` sets

These are as they sound -- the 'pool' and 'phantom' control samples are 
separated from the experimental samples. The control set is not exported 
from the `dds` object.

```{r}
# add the `purpose` field -- note that this is temporary and will be added 
# to the llfs_rnaseq_metadata view
dds_full$purpose = 
  ifelse(dds_full$visit == 0 | 
           (!is.na(dds_full$relabel_reason) & 
              dds_full$relabel_reason == 'phantom'),
         'control',
         'experiment')

# some checks on the metadata -------------------------------------------------
## ensure that there are no mislabel TRUE that have been relabelled already
stopifnot(ncol(dds_full[,!is.na(dds_full$relabel_reason) & 
                          dds_full$mislabel]) == 0)

stopifnot(unique(dds_full$legal) == 'yes')

# split into control and experimental sets ------------------------------------
dds_control = dds_full[,dds_full$purpose == 'control']
dds_experiment = dds_full[,dds_full$purpose == 'experiment']
```

## Create a gene-wise (expression) filter

```{r}
# Filtering -------------------------------------------------------------------

## Expression Filter ----

# at the time of writing, num_samples comes out to 19
num_samples = floor(ncol(dds_experiment)*.015)

# filter out low expression genes
mid_expression_filter <- rowSums(cpm(counts(dds_experiment))  > 3) >= num_samples
```

## Create a sample-wise filter

We wish to filter out both (subject, visit) duplicates -- this can arise due 
to either on-purpose re-dos of "bad" samples, or due to accidental mislabels 
which create duplicate samples -- and those samples which fail QC. 

```{r}

## Deduplicate
## duplications can occur due to sample re-dos, or due to relabelling
## mislabelled samples
dedup_sample_ids = colData(dds_experiment) %>%
  as_tibble() %>%
  group_by(subject, visit) %>%
  filter(percent_intergenic == min(percent_intergenic)) %>%
  pull(sample_id)

## Quality Filter ----

# passing samples have less than 8% intergenic reads, do not express
# the wrong sex chromosome according to their sample label, and are not
# identified as mislabels from the RNA variant vs WGS DNA variant comparison.
# Note that a sample labelled "mislabelled" is one whose variants clearly 
# mismatched the WGS sample of the same subject identifier, but for which 
# there was no obvious match also found in the WGS subject set
passing_qc_sample_ids = colData(dds_experiment) %>%
  as_tibble() %>%
  # note the negation at the beginning of the conditional -- I find it easier
  # to think about samples which are failing rather than the other way around
  filter(!mislabelled, !suspicious_sex, percent_intergenic < 0.08) %>%
  pull(sample_id)

analysis_sample_ids = intersect(dedup_filter, qc_filter)
```

## Apply the filters and export plain text data

```{r}
## Apply Filters ----

# filter the deseq data object on the rows (genes) and samples (column)
dds_experiment_passing = 
  dds_experiment[mid_expression_filter, 
                 dds_experiment$sample_id %in% analysis_sample_ids]

# add size factors to the dds object
dds_experiment_passing = estimateSizeFactors(dds_experiment_passing)

# extract data from dds object ------------------------------------------------

output_list = list(
  raw_unfltr_counts = as_tibble(counts(dds_full)),
  unfltr_feature_id = tibble(gene_id = rownames(dds_full)),
  raw_fltr_counts = as_tibble(counts(dds_experiment_passing)),
  norm_fltr_counts = as_tibble(counts(dds_experiment_passing, normalized=TRUE)),
  fltr_feature_id = tibble(gene_id = rownames(dds_experiment_passing))
)

# write out -------------------------------------------------------------------

if(WRITE){
  today = format(Sys.time(), "%Y%m%d")

  if(TX_QUANTS){
    add_sep = "_tx_"
  } else{
    add_sep = "_gene_"
  }

  sample_metadata_path = here("llfs_rnaseq_data/plain_text", 
                              paste0("sample_meta_", 
                                     today, ".csv"))
  # Do not overwrite -- if this is already been run, ie if you run the gene
  # data and then the tx data, there is no reason to re-write the sample
  # metadata. Pay attention to this if you do want to re-write on the same
  # day
  if(!file.exists(sample_metadata_path)){
    write_csv(as_tibble(colData(dds_full)), sample_metadata_path)
  }

  map(names(output_list),
      ~write_csv(output_list[[.]],
                 file.path(here("llfs_rnaseq_data/plain_text"),
                           paste0(., add_sep, today, ".csv"))))
}
```
