---
title: "shared-data-overview"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{shared-data-overview}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  eval=FALSE
)
```

# llfs_shared_data_<version>

This top level directory now contains the objects which themselves contain the 
shared data. The subdirectory `plain_text` now stores the `.csv` files which 
are extracted from these objects. For a more attractive view of this 
information, see 
[here](file:///home/oguzkhan/code/llfsRnaseq/docs/articles/shared-data-overview.html)

## llfs_rnaseq_database.sqlite

This is a SQLite database which almost entirely functions to assign a unique 
ID to a specific biological sample which is then sequenced. The reason an 
invariable primary key has become important for biological samples is that we 
are re-labeling a number of samples (meaning, assigning a different ID/subject) 
for different reasons. Without a unique and invariable primary key across 
releases, it becomes extremely difficult to understand how and why a sample 
might behave differently. And, more importantly, it becomes even more difficult 
to relate a given sample in the cleaned, relabelled, shared data with the 
fastq file from which it originated (since the fastq file is labelled with 
a different ID).  

The database is fully defined and described 
[here](https://cmatkhan.github.io/llfsRnaseq/articles/database-overview.html)

## dds_analysis_<date>_<gene,tx>.rds

a Deseq Data Object with Granges in the rows (features), sample metadata in
the columns, and the count data in the assays slot.

There are two -- the one with the suffix _gene.rds has counts collapsed to genes.
The one with _tx.rds has counts by transcript.

These objects are used to create the `.csv` files in the subdirectory 
`plain_text`. 
See [here](https://cmatkhan.github.io/llfsRnaseq/articles/creating-a-release.html) 
for details on how.

## plain_text

This subdirectory stores `.csv` files, extracted from the `.rds` objects,
of the counts and metadata.

# plain_text

This is a subdirectory of the `llfs_shared_data_<version>` directory. It stores 
`.csv` files extracted from the objects in the parent directory. 

## Things to note

The difference between unfltr (unfiltered) and fltr (filtered) gene and tx
counts is the application of a gene (row) filter on gene expression,
and sample (column) filter on sample quality. Genes pass the gene expression
filter if a given gene has greater than or equal to 3 counts per million in
at least 98.5% of the samples. The sample quality filter is described in more
detail in the sample_metadata_<date>.csv section below. You can see how 
these filters are applied in the vignette 
[Setting Filters and sharing data](https://cmatkhan.github.io/llfsRnaseq/articles/creating-a-release.html)

Below is a description of each file:

## sample_metadata_<date>.csv

See 
[llfs_rnaseq_metadata](https://cmatkhan.github.io/llfsRnaseq/articles/database-overview.html#llfs_rnaseq_metadata) 
for details on what each variable in the metadata means.

## unfltr_<gene,tx>\_id\_<date>.csv

This corresponds to the rows of the __unfiltered__ counts.
These originate from the gene_id or transcript_id of
[gencode38](https://www.gencodegenes.org/human/release_38.html).

## fltr_<gene,tx>\_id\_<date>.csv

This corresponds to the rows of the __filtered__ counts. These originate from
[gencode38](https://www.gencodegenes.org/human/release_38.html)

## raw_unfltr_counts_<gene,tx>\_<date>.csv

the 'count' matrix that results from tximport (see Overview) -- all genes,
all samples

## raw_fltr_counts_<gene,tx>\_<date>.csv

Raw 'counts' with both expression and sample filters applied. 
See R/extract_data_from_dds.R for details.

## norm_fltr_counts_<gene,tx>\_<date>.csv

Normalization 'counts' with both expression and sample filters applied. 
See R/extract_data_from_dds.R for details. Normaliation method:  

DESeq size factor (see the DESeq2 paper for details) and average transcript
length normalized.

