<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>database-overview • llfsRnaseq</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="database-overview">
<meta property="og:description" content="llfsRnaseq">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">llfsRnaseq</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/llfsRnaseq.html">Get started</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Vignettes
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="dropdown-header">The RNAseq Database</li>
    <li>
      <a href="../articles/database-overview.html">Overview</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Processing Raw Data</li>
    <li>
      <a href="../articles/create-samplesheet.html">Creating A Samplesheet</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Initial QC</li>
    <li>
      <a href="../articles/sex-label-check.html">Sex Label Check</a>
    </li>
    <li>
      <a href="../articles/expression-qc.html">Expression QC</a>
    </li>
    <li>
      <a href="../articles/batch-effects.html">Batch Effects</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Creating a release</li>
    <li>
      <a href="../articles/creating-a-release.html">Setting filters and Exporting the Data</a>
    </li>
    <li>
      <a href="../articles/shared-data-overview.html">Shared Data Overview</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">API</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
<li>
  <a href="https://nf-co.re/rnaseq/3.3/" class="external-link">nf-core/rnaseq pipeline</a>
</li>
<li>
  <a href="https://github.com/cmatKhan/rnavar" class="external-link">modified nf-core/rnavar</a>
</li>
<li>
  <a href="https://longlifefamilystudy.wustl.edu" class="external-link"></a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/cmatKhan/llfsRnaseq">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>database-overview</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/cmatKhan/llfsRnaseq/blob/HEAD/vignettes/database-overview.Rmd"><code>vignettes/database-overview.Rmd</code></a></small>
      <div class="hidden name"><code>database-overview.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="tldr">tl;dr<a class="anchor" aria-label="anchor" href="#tldr"></a>
</h2>
<p>This database will make re-labelling samples less error prone, and
will make it much easier to submit to NCBI when the time comes. For each
<strong>table</strong> (not view) below, there is a brief description of
the table, and the primary and foreign key(s) are shown. A full table
description with all of the fields are available if you click the
details for each table and view.</p>
</div>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>This is a database whose purpose is to assign a unique ID to both
“samples”, which can be thought of as a specific tube of blood donated
by a given LLFS participant during a given visit, and to individual
fastq files themselves. Note that one “sample” might have up to six or
more (should be even, of course) fastq files – the paired-end R1 and R2,
and sometimes a single sample has multiple R1 and R2 files over
different lanes.</p>
<p>Prior to the July, 2023 release, this data was being tracked in a
series of excel files. It was challenging to keep track of what file was
what as they were amended and passed back and forth. Since some of these
files pertained to relabeling samples due to typos, or the result of
comparing the RNA variants to the WGS data, and there did not exist a
unique sample ID other than the combination of (subject, visit, batch),
which was of course mutable since <code>subject</code> would need to be
changed in the event of a mislabel, a database became a necessity.</p>
<p>The result of using a database means that a sample is assigned an
immutable primary key, the <a href="#sample">sample table</a> primary
key, frequently referred to in other tables as <code>sample_id</code>.
Changing any other labeling, for instance the <code>whatdatall</code>
<code>identifier</code> and/or <code>subject</code>, becomes a simple
task. Moreover, this means that by changing the <code>identifier</code>,
you also change the <code>sex</code> – no more having to change two
columns for every mislabel, a source of potential error that is easily
avoided in a database.</p>
</div>
<div class="section level2">
<h2 id="important">IMPORTANT<a class="anchor" aria-label="anchor" href="#important"></a>
</h2>
<p>Please note the following:</p>
<p>All tables have an auto-incremented primary key called
<code>pk</code>. This is because any column called <code>id</code> is
specifically referring to the LLFS whatdatall <code>id</code>. However,
when a field is a foreign key, I give it the suffix <code>_id</code>. So
for example, <code>sample_id</code> is a foreign key to the
<code>sample</code> table.</p>
<p>This is not great – any specific suggestions on how to improve this
naming would be appreciated.</p>
<p>To make this more complicated, <code>fastq_id</code> is always a
foreign key to the <code>fastq</code> table, but it has an additional
meaning: the <code>fastq_id</code> is the <code>id</code> which the
actual fastq file possesses. This <code>fastq_id</code> may be wrong –
for example, if there was a typo, or that particular ID is in the
visit_1 transfer set, or if it was determined to be mislabeled from the
RNA variants vs WGS data comparison. In this case, a sample should be
thought of as having a <code>fastq_id</code>, derived from the
fastqfilename, but also a <code>whatdatall_id</code>, which results from
a correction (<a href="#corrected_sample">see the corrected_samples
table</a>). a given sample’s <code>subject</code> will always be
assigned from the samples <code>whatdatall_id</code>, which, if you are
using one of my views, is assigned after relabeling any known
mislabels.</p>
</div>
<div class="section level2">
<h2 id="views">Views<a class="anchor" aria-label="anchor" href="#views"></a>
</h2>
<p>A view is a saved SQL query. It provides a way to access joined
and/or otherwise manipulated data as if it were a table.</p>
<div class="section level3">
<h3 id="llfs_rnaseq_metadata">llfs_rnaseq_metadata<a class="anchor" aria-label="anchor" href="#llfs_rnaseq_metadata"></a>
</h3>
<p>This provides the metadata which I use when constructing the released
data. Note that not all of these columns are provided in the metadata of
the released data</p>
<details><summary>
LLFS RNAseq Metadata View
</summary><table class="table">
<colgroup>
<col width="16%">
<col width="18%">
<col width="9%">
<col width="56%">
</colgroup>
<thead><tr class="header">
<th>Field</th>
<th>Purpose</th>
<th>Datatype</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>sample_id</td>
<td></td>
<td>INTEGER</td>
<td>Primary key of the sample</td>
</tr>
<tr class="even">
<td>batch_id</td>
<td></td>
<td>INTEGER</td>
<td>Foreign key to the batch table</td>
</tr>
<tr class="odd">
<td>batch_alias</td>
<td></td>
<td>TEXT</td>
<td>Alias of the batch</td>
</tr>
<tr class="even">
<td>fastq_id</td>
<td></td>
<td>TEXT</td>
<td>Foreign key to the fastq table</td>
</tr>
<tr class="odd">
<td>visit</td>
<td></td>
<td>TEXT</td>
<td>Visit information for the sample</td>
</tr>
<tr class="even">
<td>relabel_reason</td>
<td></td>
<td>TEXT</td>
<td>Reason for relabeling the sample (or “NA” if not applicable)</td>
</tr>
<tr class="odd">
<td>mislabelled</td>
<td></td>
<td>INTEGER</td>
<td>Indicator of whether the sample is mislabeled (0 or 1)</td>
</tr>
<tr class="even">
<td>suspicious_sex</td>
<td></td>
<td>INTEGER</td>
<td>Indicator of whether the sample has suspicious sex (0 or 1)</td>
</tr>
<tr class="odd">
<td>legal</td>
<td></td>
<td>TEXT</td>
<td>Reason for legal status (“yes” if not applicable)</td>
</tr>
<tr class="even">
<td>whatdatall_id</td>
<td></td>
<td>INTEGER</td>
<td>Foreign key to the whatdatall table</td>
</tr>
<tr class="odd">
<td>subject</td>
<td></td>
<td>INTEGER</td>
<td>Subject information for the sample</td>
</tr>
<tr class="even">
<td>sex</td>
<td></td>
<td>INTEGER</td>
<td>Sex information for the sample</td>
</tr>
<tr class="odd">
<td>chromosome_x</td>
<td></td>
<td>INTEGER</td>
<td>Number of reads mapped to chromosome X</td>
</tr>
<tr class="even">
<td>chromosome_y</td>
<td></td>
<td>INTEGER</td>
<td>Number of reads mapped to chromosome Y</td>
</tr>
<tr class="odd">
<td>y_x_ratio</td>
<td></td>
<td>REAL</td>
<td>Ratio of reads mapped to chromosome Y to chromosome X</td>
</tr>
<tr class="even">
<td>exonic</td>
<td></td>
<td>INTEGER</td>
<td>Number of reads in exonic regions</td>
</tr>
<tr class="odd">
<td>intronic</td>
<td></td>
<td>INTEGER</td>
<td>Number of reads in intronic regions</td>
</tr>
<tr class="even">
<td>intergenic</td>
<td></td>
<td>INTEGER</td>
<td>Number of reads in intergenic regions</td>
</tr>
<tr class="odd">
<td>percent_exonic</td>
<td></td>
<td>REAL</td>
<td>Percentage of reads in exonic regions</td>
</tr>
<tr class="even">
<td>percent_intronic</td>
<td></td>
<td>REAL</td>
<td>Percentage of reads in intronic regions</td>
</tr>
<tr class="odd">
<td>percent_intergenic</td>
<td></td>
<td>REAL</td>
<td>Percentage of reads in intergenic regions</td>
</tr>
</tbody>
</table></details>
</div>
<div class="section level3">
<h3 id="genomic_origin_view">genomic_origin_view<a class="anchor" aria-label="anchor" href="#genomic_origin_view"></a>
</h3>
<p>This view provides information about the genomic origin of reads
based on the <code>qualimap_genomic_origin</code> table.</p>
<details><summary>
Qualimap Genomic Origin View
</summary><div class="line-block">Field | Purpose | Datatype | Description |</div>
|——————–|————————–|———-|———————————————————————-| | | sample_id | | TEXT
| Foreign key to the sample table | | exonic | | INTEGER | Number of
reads in exonic regions | | intronic | | INTEGER | Number of reads in
intronic regions | | intergenic | | INTEGER | Number of reads in
intergenic regions | | percent_exonic | | REAL | Percentage of reads in
exonic regions | | percent_intronic | | REAL | Percentage of reads in
intronic regions | | percent_intergenic | | REAL | Percentage of reads
in intergenic regions |
</details>
</div>
<div class="section level3">
<h3 id="xy_ratio">xy_ratio<a class="anchor" aria-label="anchor" href="#xy_ratio"></a>
</h3>
<p>This view provides information about the XY ratio based on the
<code>samtools_idxstats_xy_plot</code> table.</p>
<details><summary>
XY Ratio View
</summary><table class="table">
<colgroup>
<col width="15%">
<col width="28%">
<col width="10%">
<col width="45%">
</colgroup>
<thead><tr class="header">
<th>Field</th>
<th>Purpose</th>
<th>Datatype</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>sample_id</td>
<td></td>
<td>TEXT</td>
<td>Foreign key to the sample table</td>
</tr>
<tr class="even">
<td>chromosome_x</td>
<td></td>
<td>INTEGER</td>
<td>Number of reads mapped to chromosome X</td>
</tr>
<tr class="odd">
<td>chromosome_y</td>
<td></td>
<td>INTEGER</td>
<td>Number of reads mapped to chromosome Y</td>
</tr>
<tr class="even">
<td>y_x_ratio</td>
<td></td>
<td>REAL</td>
<td>Ratio of reads mapped to Y over reads mapped to X</td>
</tr>
</tbody>
</table></details>
</div>
<div class="section level3">
<h3 id="expression_qc_view">expression_qc_view<a class="anchor" aria-label="anchor" href="#expression_qc_view"></a>
</h3>
<p>This is a view of the expression_qc table with additional columns
created from the hemoglobin, rn7sl and protein_coding_total / library
size</p>
<details><summary>
Full Table Details
</summary><table class="table">
<colgroup>
<col width="19%">
<col width="21%">
<col width="8%">
<col width="50%">
</colgroup>
<thead><tr class="header">
<th>Field</th>
<th>Purpose</th>
<th>Datatype</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>pk</td>
<td>primary key autoincrement</td>
<td>INTEGER</td>
<td>auto incremented primary key</td>
</tr>
<tr class="even">
<td>sample_id</td>
<td>foreign key</td>
<td>INTEGER</td>
<td>References the “pk” field in the “sample” table</td>
</tr>
<tr class="odd">
<td>library_size</td>
<td></td>
<td>INTEGER</td>
<td>Holds the library size data for each record</td>
</tr>
<tr class="even">
<td>protein_coding_total</td>
<td></td>
<td>INTEGER</td>
<td>Represents the total count of protein coding for each record</td>
</tr>
<tr class="odd">
<td>hemoglobin_counts</td>
<td></td>
<td>INTEGER</td>
<td>Stores the count of hemoglobin for each record</td>
</tr>
<tr class="even">
<td>rn7sl1_counts</td>
<td></td>
<td>INTEGER</td>
<td>Stores the count of rn7sl1 for each record</td>
</tr>
<tr class="odd">
<td>percent_hemoglobin</td>
<td></td>
<td>INTEGER</td>
<td>hemoglobin_counts / library_size</td>
</tr>
<tr class="even">
<td>percent_rn7sl</td>
<td></td>
<td>INTEGER</td>
<td>rn7sl_counts / library_size</td>
</tr>
<tr class="odd">
<td>percent_protein_coding</td>
<td></td>
<td>INTEGER</td>
<td>protein_coding_total / library_size</td>
</tr>
</tbody>
</table></details>
</div>
</div>
<div class="section level2">
<h2 id="tables">Tables<a class="anchor" aria-label="anchor" href="#tables"></a>
</h2>
<p>Each heading below is a physical table (not a view) in the database.
A brief description of each is provided, and the primary key and foreign
key, if one exists, is shown to give you an overview of how these tables
are related.</p>
<p>To see the full description of any one of these tables, click “Full
Table Details”.</p>
<div class="section level3">
<h3 id="batch">batch<a class="anchor" aria-label="anchor" href="#batch"></a>
</h3>
<p>This table stores information about batches.</p>
<table class="table">
<thead><tr class="header">
<th>Field</th>
<th>Purpose</th>
<th>Datatype</th>
<th>Description</th>
</tr></thead>
<tbody><tr class="odd">
<td>pk</td>
<td>primary key</td>
<td>INTEGER</td>
<td>autoincremented primary key</td>
</tr></tbody>
</table>
<details><summary>
Full Table Details
</summary><table class="table">
<colgroup>
<col width="17%">
<col width="23%">
<col width="25%">
<col width="33%">
</colgroup>
<thead><tr class="header">
<th>Field</th>
<th>Purpose</th>
<th>Datatype</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>pk</td>
<td>primary key</td>
<td>INTEGER</td>
<td>autoincremented primary key</td>
</tr>
<tr class="even">
<td>data_dir</td>
<td></td>
<td>TEXT</td>
<td>name of the directory which stores this data in the dsg cluster</td>
</tr>
<tr class="odd">
<td>batch_alias</td>
<td></td>
<td>TEXT</td>
<td>I was unfortunately inconsistent in how I refered to these in
processing the data. this column provides a map from what I named the
batch when processing to the name of the dsg directory</td>
</tr>
<tr class="even">
<td>dsg_fastq_dirpath</td>
<td></td>
<td>TEXT</td>
<td>The subdirectory structure of each data_dir varies quite a bit. This
column provides the absolute path to the actual directory which stores
the fastq files</td>
</tr>
</tbody>
</table></details>
</div>
<div class="section level3">
<h3 id="corrected_sample">corrected_sample<a class="anchor" aria-label="anchor" href="#corrected_sample"></a>
</h3>
<p>This table stores information about corrected samples. Note that only
<code>legal == 'yes'</code> samples are present in this table</p>
<table class="table">
<colgroup>
<col width="17%">
<col width="23%">
<col width="25%">
<col width="33%">
</colgroup>
<thead><tr class="header">
<th>Field</th>
<th>Purpose</th>
<th>Datatype</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>pk</td>
<td>primary key</td>
<td>INTEGER</td>
<td>autoincremented primary key</td>
</tr>
<tr class="even">
<td>sample_id</td>
<td>foreign key to sample</td>
<td>TEXT</td>
<td>References the “pk” field in the “sample” table</td>
</tr>
</tbody>
</table>
<details><summary>
Full Table Details
</summary><table class="table">
<colgroup>
<col width="17%">
<col width="23%">
<col width="25%">
<col width="33%">
</colgroup>
<thead><tr class="header">
<th>Field</th>
<th>Purpose</th>
<th>Datatype</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>pk</td>
<td>primary key</td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="even">
<td>sample_id</td>
<td>foreign key to sample</td>
<td>TEXT</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="odd">
<td>whatdatall_id</td>
<td>foreign key to whatdatall</td>
<td>TEXT</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="even">
<td>reason</td>
<td></td>
<td>TEXT</td>
<td>Add a brief explanation of the field here.</td>
</tr>
</tbody>
</table></details>
</div>
<div class="section level3">
<h3 id="expression_qc">expression_qc<a class="anchor" aria-label="anchor" href="#expression_qc"></a>
</h3>
<p>This table stores count data which is extracted in the expression_qc
step (see the articles in the served documentation, or the vignette in
the code repo).</p>
<table class="table">
<colgroup>
<col width="19%">
<col width="21%">
<col width="8%">
<col width="50%">
</colgroup>
<thead><tr class="header">
<th>Field</th>
<th>Purpose</th>
<th>Datatype</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>pk</td>
<td>primary key autoincrement</td>
<td>INTEGER</td>
<td>auto incremented primary key</td>
</tr>
<tr class="even">
<td>sample_id</td>
<td>foreign key</td>
<td>INTEGER</td>
<td>References the “pk” field in the “sample” table</td>
</tr>
</tbody>
</table>
<details><summary>
Full Table Details
</summary><table class="table">
<colgroup>
<col width="19%">
<col width="21%">
<col width="8%">
<col width="50%">
</colgroup>
<thead><tr class="header">
<th>Field</th>
<th>Purpose</th>
<th>Datatype</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>pk</td>
<td>primary key autoincrement</td>
<td>INTEGER</td>
<td>auto incremented primary key</td>
</tr>
<tr class="even">
<td>sample_id</td>
<td>foreign key</td>
<td>INTEGER</td>
<td>References the “pk” field in the “sample” table</td>
</tr>
<tr class="odd">
<td>library_size</td>
<td></td>
<td>INTEGER</td>
<td>Holds the library size data for each record</td>
</tr>
<tr class="even">
<td>protein_coding_total</td>
<td></td>
<td>INTEGER</td>
<td>Represents the total count of protein coding for each record</td>
</tr>
<tr class="odd">
<td>hemoglobin_counts</td>
<td></td>
<td>INTEGER</td>
<td>Stores the count of hemoglobin for each record</td>
</tr>
<tr class="even">
<td>rn7sl1_counts</td>
<td></td>
<td>INTEGER</td>
<td>Stores the count of rn7sl1 for each record</td>
</tr>
</tbody>
</table></details>
</div>
<div class="section level3">
<h3 id="fastq">fastq<a class="anchor" aria-label="anchor" href="#fastq"></a>
</h3>
<p>This table stores information about FASTQ files. Note that there may
be more than 1 fastq record for each sample_id since there are some
(quite a few) samples which have multiple lanes worth of paired-end
libraries (so if 1 sample is split across two lanes, there would be 2 R1
and 2 R2)</p>
<table class="table">
<colgroup>
<col width="17%">
<col width="23%">
<col width="25%">
<col width="33%">
</colgroup>
<thead><tr class="header">
<th>Field</th>
<th>Purpose</th>
<th>Datatype</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>pk</td>
<td>primary key</td>
<td>INTEGER</td>
<td>auto incremented primary key</td>
</tr>
<tr class="even">
<td>sample_id</td>
<td>foreign key to sample</td>
<td>TEXT</td>
<td>References the “pk” field in the “sample” table</td>
</tr>
</tbody>
</table>
<details><summary>
Full Table Details
</summary><table class="table">
<colgroup>
<col width="17%">
<col width="23%">
<col width="25%">
<col width="33%">
</colgroup>
<thead><tr class="header">
<th>Field</th>
<th>Purpose</th>
<th>Datatype</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>pk</td>
<td>primary key</td>
<td>INTEGER</td>
<td>autoincremented primary key</td>
</tr>
<tr class="even">
<td>sample_id</td>
<td>foreign key to sample</td>
<td>INTEGER</td>
<td>foreign key to the sample table</td>
</tr>
<tr class="odd">
<td>fastq_1</td>
<td></td>
<td>TEXT</td>
<td>name of the R1 fastq file</td>
</tr>
<tr class="even">
<td>fastq_2</td>
<td></td>
<td>TEXT</td>
<td>name of the R2 fastq file</td>
</tr>
<tr class="odd">
<td>strandedness</td>
<td></td>
<td>TEXT</td>
<td>the strandedness of the library protocol</td>
</tr>
<tr class="even">
<td>index_sequence</td>
<td></td>
<td>TEXT</td>
<td>index sequence extracted from the fastq filename</td>
</tr>
<tr class="odd">
<td>flow_cell_id</td>
<td>foreign key to flow_cell</td>
<td>TEXT</td>
<td>flow cell identifier</td>
</tr>
<tr class="even">
<td>lane</td>
<td></td>
<td>TEXT</td>
<td>lane in the flowcell</td>
</tr>
<tr class="odd">
<td>notes</td>
<td></td>
<td>TEXT</td>
<td>any notes on the fastq. Defaults to ‘none’</td>
</tr>
</tbody>
</table></details>
</div>
<div class="section level3">
<h3 id="id_correction">id_correction<a class="anchor" aria-label="anchor" href="#id_correction"></a>
</h3>
<p>This table stores a list of the possible reasons why a sample’s
whatdatall ID might be corrected. It is used as a foreign key in
<code>corrected_samples</code> to ensure that the
<code>"corrected_samples"."reason"</code> field is consistent.</p>
<table class="table">
<colgroup>
<col width="17%">
<col width="23%">
<col width="25%">
<col width="33%">
</colgroup>
<thead><tr class="header">
<th>Field</th>
<th>Purpose</th>
<th>Datatype</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>reason</td>
<td></td>
<td>TEXT</td>
<td>reason that the sample is being mislabelled</td>
</tr>
<tr class="even">
<td>source</td>
<td></td>
<td>TEXT</td>
<td>the file or way in which the mislabel was originally
communicated</td>
</tr>
</tbody>
</table>
</div>
<div class="section level3">
<h3 id="multiqc_general">multiqc_general<a class="anchor" aria-label="anchor" href="#multiqc_general"></a>
</h3>
<p><strong>Note</strong>: parsing this data out of multiQC is a work in
progress.</p>
<p>This is the ‘general’ table stats from MultiQC and includes both
alignment and read level QC. As a result, there are at least 3 records
for each sample. This is not in its final form yet – I am still working
on the parsing. However, you can extract the data, and the data is
accurate. It is here if you are interested, and the parsing will imrpove
over time.</p>
<table class="table">
<colgroup>
<col width="17%">
<col width="23%">
<col width="25%">
<col width="33%">
</colgroup>
<thead><tr class="header">
<th>Field</th>
<th>Purpose</th>
<th>Datatype</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>pk</td>
<td>primary key</td>
<td>INTEGER</td>
<td>autoincremented primary key</td>
</tr>
<tr class="even">
<td>sample_id</td>
<td>foreign key to sample</td>
<td>INTEGER</td>
<td>References the “pk” field in the “sample” table</td>
</tr>
</tbody>
</table>
<details><summary>
Full Table Details
</summary><table class="table">
<colgroup>
<col width="17%">
<col width="23%">
<col width="25%">
<col width="33%">
</colgroup>
<thead><tr class="header">
<th>Field</th>
<th>Purpose</th>
<th>Datatype</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>pk</td>
<td>primary key</td>
<td>INTEGER</td>
<td>autoincremented primary key</td>
</tr>
<tr class="even">
<td>sample_id</td>
<td>foreign key to sample</td>
<td>INTEGER</td>
<td>References the “pk” field in the “sample” table</td>
</tr>
<tr class="odd">
<td>general.dupradar_intercept</td>
<td></td>
<td>REAL</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="even">
<td>general.percent_rrna</td>
<td></td>
<td>REAL</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="odd">
<td>general.library</td>
<td></td>
<td>TEXT</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="even">
<td>general.unpaired_reads_examined</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="odd">
<td>general.read_pairs_examined</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="even">
<td>general.secondary_or_supplementary_rds</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="odd">
<td>general.unmapped_reads</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="even">
<td>general.unpaired_read_duplicates</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="odd">
<td>general.read_pair_duplicates</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="even">
<td>general.read_pair_optical_duplicates</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="odd">
<td>general.percent_duplication</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="even">
<td>general.estimated_library_size</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="odd">
<td>general.5_3_bias</td>
<td></td>
<td>REAL</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="even">
<td>general.reads_aligned</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="odd">
<td>general.unalignable</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="even">
<td>general.alignable</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="odd">
<td>general.filtered</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="even">
<td>general.total</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="odd">
<td>general.alignable_percent</td>
<td></td>
<td>REAL</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="even">
<td>general.unique</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="odd">
<td>general.multi</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="even">
<td>general.uncertain</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="odd">
<td>general.total_records</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="even">
<td>general.qc_failed</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="odd">
<td>general.optical_pcr_duplicate</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="even">
<td>general.non_primary_hits</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="odd">
<td>general.mapq_lt_mapq_cut_non_unique</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="even">
<td>general.mapq_gte_mapq_cut_unique</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="odd">
<td>general.read_1</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="even">
<td>general.read_2</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="odd">
<td>general.reads_map_to_sense</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="even">
<td>general.reads_map_to_antisense</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="odd">
<td>general.non_splice_reads</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="even">
<td>general.splice_reads</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="odd">
<td>general.reads_mapped_in_proper_pairs</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="even">
<td>general.proper_paired_reads_map_to_different_chrom</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="odd">
<td>general.unique_percent</td>
<td></td>
<td>REAL</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="even">
<td>general.proper_pairs_percent</td>
<td></td>
<td>REAL</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="odd">
<td>general.raw_total_sequences</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="even">
<td>general.filtered_sequences</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="odd">
<td>general.sequences</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="even">
<td>general.is_sorted</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="odd">
<td>general.1st_fragments</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="even">
<td>general.last_fragments</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="odd">
<td>general.reads_mapped</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="even">
<td>general.reads_mapped_and_paired</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="odd">
<td>general.reads_unmapped</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="even">
<td>general.reads_properly_paired</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="odd">
<td>general.reads_paired</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="even">
<td>general.reads_duplicated</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="odd">
<td>general.reads_mq0</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="even">
<td>general.reads_qc_failed</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="odd">
<td>general.non_primary_alignments</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="even">
<td>general.supplementary_alignments</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="odd">
<td>general.total_length</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="even">
<td>general.total_first_fragment_length</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="odd">
<td>general.total_last_fragment_length</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="even">
<td>general.bases_mapped</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="odd">
<td>general.bases_mapped_cigar</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="even">
<td>general.bases_trimmed</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="odd">
<td>general.bases_duplicated</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="even">
<td>general.mismatches</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="odd">
<td>general.error_rate</td>
<td></td>
<td>REAL</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="even">
<td>general.average_length</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="odd">
<td>general.average_first_fragment_length</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="even">
<td>general.average_last_fragment_length</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="odd">
<td>general.maximum_length</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="even">
<td>general.maximum_first_fragment_length</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="odd">
<td>general.maximum_last_fragment_length</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="even">
<td>general.average_quality</td>
<td></td>
<td>REAL</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="odd">
<td>general.insert_size_average</td>
<td></td>
<td>REAL</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="even">
<td>general.insert_size_standard_deviation</td>
<td></td>
<td>REAL</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="odd">
<td>general.inward_oriented_pairs</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="even">
<td>general.outward_oriented_pairs</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="odd">
<td>general.pairs_with_other_orientation</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="even">
<td>general.pairs_on_different_chromosomes</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="odd">
<td>general.percentage_of_properly_paired_reads_%</td>
<td></td>
<td>REAL</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="even">
<td>general.reads_mapped_percent</td>
<td></td>
<td>REAL</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="odd">
<td>general.reads_mapped_and_paired_percent</td>
<td></td>
<td>REAL</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="even">
<td>general.reads_unmapped_percent</td>
<td></td>
<td>REAL</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="odd">
<td>general.reads_properly_paired_percent</td>
<td></td>
<td>REAL</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="even">
<td>general.reads_paired_percent</td>
<td></td>
<td>REAL</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="odd">
<td>general.reads_duplicated_percent</td>
<td></td>
<td>REAL</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="even">
<td>general.reads_mq0_percent</td>
<td></td>
<td>REAL</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="odd">
<td>general.reads_qc_failed_percent</td>
<td></td>
<td>REAL</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="even">
<td>general.total_passed</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="odd">
<td>general.total_failed</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="even">
<td>general.secondary_passed</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="odd">
<td>general.secondary_failed</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="even">
<td>general.supplementary_passed</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="odd">
<td>general.supplementary_failed</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="even">
<td>general.duplicates_passed</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="odd">
<td>general.duplicates_failed</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="even">
<td>general.mapped_passed</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="odd">
<td>general.mapped_failed</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="even">
<td>general.mapped_passed_pct</td>
<td></td>
<td>REAL</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="odd">
<td>general.mapped_failed_pct</td>
<td></td>
<td>REAL</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="even">
<td>general.paired_in_sequencing_passed</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="odd">
<td>general.paired_in_sequencing_failed</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="even">
<td>general.read1_passed</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="odd">
<td>general.read1_failed</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="even">
<td>general.read2_passed</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="odd">
<td>general.read2_failed</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="even">
<td>general.properly_paired_passed</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="odd">
<td>general.properly_paired_failed</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="even">
<td>general.properly_paired_passed_pct</td>
<td></td>
<td>REAL</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="odd">
<td>general.properly_paired_failed_pct</td>
<td></td>
<td>REAL</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="even">
<td>general.with_itself_and_mate_mapped_passed</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="odd">
<td>general.with_itself_and_mate_mapped_failed</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="even">
<td>general.singletons_passed</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="odd">
<td>general.singletons_failed</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="even">
<td>general.singletons_passed_pct</td>
<td></td>
<td>REAL</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="odd">
<td>general.singletons_failed_pct</td>
<td></td>
<td>REAL</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="even">
<td>general.with_mate_mapped_to_a_different_chr_passed</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="odd">
<td>general.with_mate_mapped_to_a_different_chr_failed</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="even">
<td>general.with_mate_mapped_to_a_different_chr_mapq__5_passed</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="odd">
<td>general.with_mate_mapped_to_a_different_chr_mapq__5_failed</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="even">
<td>general.flagstat_total</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="odd">
<td>general.percent_gc</td>
<td></td>
<td>REAL</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="even">
<td>general.avg_sequence_length</td>
<td></td>
<td>REAL</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="odd">
<td>general.total_sequences</td>
<td></td>
<td>REAL</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="even">
<td>general.percent_duplicates</td>
<td></td>
<td>REAL</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="odd">
<td>general.percent_fails</td>
<td></td>
<td>REAL</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="even">
<td>general.r_processed</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="odd">
<td>general.r_with_adapters</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="even">
<td>general.r_written</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="odd">
<td>general.bp_processed</td>
<td></td>
<td>REAL</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="even">
<td>general.quality_trimmed</td>
<td></td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="odd">
<td>general.bp_written</td>
<td></td>
<td>REAL</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="even">
<td>general.percent_trimmed</td>
<td></td>
<td>REAL</td>
<td>Add a brief explanation of the field here.</td>
</tr>
</tbody>
</table></details>
</div>
<div class="section level3">
<h3 id="qualimap_genomic_origin">qualimap_genomic_origin<a class="anchor" aria-label="anchor" href="#qualimap_genomic_origin"></a>
</h3>
<p>This table stores genomic origin information from QualiMap. See the
QualiMap documentation for more details on how a read is assigned to
each category.</p>
<table class="table">
<colgroup>
<col width="17%">
<col width="23%">
<col width="25%">
<col width="33%">
</colgroup>
<thead><tr class="header">
<th>Field</th>
<th>Purpose</th>
<th>Datatype</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>pk</td>
<td>primary key</td>
<td>INTEGER</td>
<td>autoincremented primary key</td>
</tr>
<tr class="even">
<td>sample_id</td>
<td>foreign key to sample</td>
<td>TEXT</td>
<td>References the “pk” field in the “sample” table</td>
</tr>
</tbody>
</table>
<details><summary>
Full Table Details
</summary><table class="table">
<colgroup>
<col width="17%">
<col width="23%">
<col width="25%">
<col width="33%">
</colgroup>
<thead><tr class="header">
<th>Field</th>
<th>Purpose</th>
<th>Datatype</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>pk</td>
<td>primary key</td>
<td>INTEGER</td>
<td>autoincremented primary key</td>
</tr>
<tr class="even">
<td>sample_id</td>
<td>foreign key to sample</td>
<td>TEXT</td>
<td>References the “pk” field in the “sample” table</td>
</tr>
<tr class="odd">
<td>exonic</td>
<td></td>
<td>INTEGER</td>
<td>raw counts of reads over exonic regions</td>
</tr>
<tr class="even">
<td>intronic</td>
<td></td>
<td>INTEGER</td>
<td>raw counts of reads over intronic regions</td>
</tr>
<tr class="odd">
<td>intergenic</td>
<td></td>
<td>INTEGER</td>
<td>raw counts of reads over intergenic regions</td>
</tr>
</tbody>
</table></details>
</div>
<div class="section level3">
<h3 id="remove_ids">remove_ids<a class="anchor" aria-label="anchor" href="#remove_ids"></a>
</h3>
<p>This table stores information about removed IDs. Note that if there
is ever more than 1 reason (currently the only reason is
<code>legal_removal</code>), then the reasons/sources should be
extracted to a separate table like <code>id_correction</code></p>
<table class="table">
<colgroup>
<col width="17%">
<col width="23%">
<col width="25%">
<col width="33%">
</colgroup>
<thead><tr class="header">
<th>Field</th>
<th>Purpose</th>
<th>Datatype</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>pk</td>
<td>primary key</td>
<td>INTEGER</td>
<td>autoincremented primary key</td>
</tr>
<tr class="even">
<td>id</td>
<td></td>
<td>INTEGER</td>
<td>References the “id” field in the “whatdatall” table</td>
</tr>
</tbody>
</table>
<details><summary>
Full Table Details
</summary><table class="table">
<colgroup>
<col width="17%">
<col width="23%">
<col width="25%">
<col width="33%">
</colgroup>
<thead><tr class="header">
<th>Field</th>
<th>Purpose</th>
<th>Datatype</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>pk</td>
<td>primary key</td>
<td>INTEGER</td>
<td>autoincremented primary key</td>
</tr>
<tr class="even">
<td>id</td>
<td></td>
<td>INTEGER</td>
<td>References the “id” field in the “whatdatall” table</td>
</tr>
<tr class="odd">
<td>source</td>
<td></td>
<td>TEXT</td>
<td>What document, or what email, etc was used to communicate that this
sample should be removed</td>
</tr>
<tr class="even">
<td>reason</td>
<td></td>
<td>TEXT</td>
<td>Reason that the sample is being removed</td>
</tr>
</tbody>
</table></details>
</div>
<div class="section level3">
<h3 id="rsem_assignment_plot">rsem_assignment_plot<a class="anchor" aria-label="anchor" href="#rsem_assignment_plot"></a>
</h3>
<p>This table stores RSEM quantification QC information.</p>
<table class="table">
<colgroup>
<col width="17%">
<col width="23%">
<col width="25%">
<col width="33%">
</colgroup>
<thead><tr class="header">
<th>Field</th>
<th>Purpose</th>
<th>Datatype</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>pk</td>
<td>primary key</td>
<td>INTEGER</td>
<td>autoincremented primary key</td>
</tr>
<tr class="even">
<td>sample_id</td>
<td>foreign key to sample</td>
<td>TEXT</td>
<td>References the “pk” field in the “sample” table</td>
</tr>
</tbody>
</table>
<details><summary>
Full Table Details
</summary><table class="table">
<colgroup>
<col width="17%">
<col width="23%">
<col width="25%">
<col width="33%">
</colgroup>
<thead><tr class="header">
<th>Field</th>
<th>Purpose</th>
<th>Datatype</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>pk</td>
<td>primary key</td>
<td>INTEGER</td>
<td>autoincremented primary key</td>
</tr>
<tr class="even">
<td>sample_id</td>
<td>foreign key to sample</td>
<td>TEXT</td>
<td>References the “pk” field in the “sample” table</td>
</tr>
<tr class="odd">
<td>aligned_uniquely_to_a_gene</td>
<td></td>
<td>INTEGER</td>
<td>RSEM determined unique alignment raw counts</td>
</tr>
<tr class="even">
<td>aligned_to_multiple_genes</td>
<td></td>
<td>INTEGER</td>
<td>RSEM determined multimap raw counts</td>
</tr>
<tr class="odd">
<td>filtered_due_to_too_many_alignments</td>
<td></td>
<td>INTEGER</td>
<td>Reads not considered for the EM quantification due to ‘too many’
multimaps</td>
</tr>
<tr class="even">
<td>unalignable_reads</td>
<td></td>
<td>INTEGER</td>
<td>raw count of reads which count not be aligned in the genome</td>
</tr>
</tbody>
</table></details>
</div>
<div class="section level3">
<h3 id="rsem_quant">rsem_quant<a class="anchor" aria-label="anchor" href="#rsem_quant"></a>
</h3>
<p>This table is a counterpart to the <code>fastq</code> table, but
instead stores the <code>dsg</code> path to the RSEM quantification
files</p>
<table class="table">
<colgroup>
<col width="17%">
<col width="23%">
<col width="25%">
<col width="33%">
</colgroup>
<thead><tr class="header">
<th>Field</th>
<th>Purpose</th>
<th>Datatype</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>pk</td>
<td>primary key</td>
<td>INTEGER</td>
<td>autoincremented primary key</td>
</tr>
<tr class="even">
<td>sample_id</td>
<td>foreign key to sample</td>
<td>TEXT</td>
<td>References the “pk” field in the “sample” table</td>
</tr>
</tbody>
</table>
<details><summary>
Full Table Details
</summary><table class="table">
<colgroup>
<col width="17%">
<col width="23%">
<col width="25%">
<col width="33%">
</colgroup>
<thead><tr class="header">
<th>Field</th>
<th>Purpose</th>
<th>Datatype</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>pk</td>
<td>primary key</td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="even">
<td>sample_id</td>
<td>foreign key to sample</td>
<td>INTEGER</td>
<td>Add a brief explanation of the field here.</td>
</tr>
<tr class="odd">
<td>rsem_isoform_quant</td>
<td></td>
<td>TEXT</td>
<td>Add a brief explanation of the field here.</td>
</tr>
</tbody>
</table></details>
</div>
<div class="section level3">
<h3 id="sample">sample<a class="anchor" aria-label="anchor" href="#sample"></a>
</h3>
<p>This is a very important table. It essentially functions to assign a
unique ID to each actual <em>sample</em> (remember a sample might have
as many as 6 fastq files assigned to it because of the paired-end
nature, and the fact that a single sample can be split over mulitple
lanes). The utility of having a unique primary key for every sample is
that it gives us a way to refer to an actual biological entity
independent of the (subject,visit) indentifiers, which may be wrong.
This primary key will never change, even if the sample is re-labelled
with a different subject later on.</p>
<table class="table">
<colgroup>
<col width="17%">
<col width="23%">
<col width="25%">
<col width="33%">
</colgroup>
<thead><tr class="header">
<th>Field</th>
<th>Purpose</th>
<th>Datatype</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>pk</td>
<td>primary key</td>
<td>INTEGER</td>
<td>autoincremented primary key</td>
</tr>
<tr class="even">
<td>batch_id</td>
<td>foreign key to batch</td>
<td>INTEGER</td>
<td>References “pk” of the batch table</td>
</tr>
</tbody>
</table>
<details><summary>
Full Table Details
</summary><table class="table">
<colgroup>
<col width="17%">
<col width="23%">
<col width="25%">
<col width="33%">
</colgroup>
<thead><tr class="header">
<th>Field</th>
<th>Purpose</th>
<th>Datatype</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>pk</td>
<td>primary key</td>
<td>INTEGER</td>
<td>autoincremented primary key</td>
</tr>
<tr class="even">
<td>fastq_id</td>
<td>foreign key to fastq</td>
<td>TEXT</td>
<td>the ORIGINAL LLFS ID which is extracted from the fastq filename</td>
</tr>
<tr class="odd">
<td>visit</td>
<td></td>
<td>TEXT</td>
<td>The visit, extracted from the fastqfilename</td>
</tr>
<tr class="even">
<td>batch_id</td>
<td>foreign key to batch</td>
<td>INTEGER</td>
<td>References “pk” of the batch table</td>
</tr>
<tr class="odd">
<td>mislabelled</td>
<td></td>
<td>BOOLEAN</td>
<td>TRUE (1) if upon comparing the RNA variants to the corresponding
subject’s WGS data, the RNA sampl was obviously mislabelled. BUT, the
actual match could not be found (not all RNA samples have WGS data).
Else FALSE (0)</td>
</tr>
<tr class="even">
<td>suspicious_sex</td>
<td></td>
<td>BOOLEAN</td>
<td>TRUE (1) if the sex expression is obviously opposite of what is
labelled. A sample will</td>
</tr>
<tr class="odd">
<td>be TRUE if it a) showed suspicious sex characteristics given its
label and b) did not have data in the WGS data to do a comparison</td>
<td></td>
<td></td>
<td></td>
</tr>
</tbody>
</table></details>
</div>
<div class="section level3">
<h3 id="samtools_idxstats_xy_plot">samtools_idxstats_xy_plot<a class="anchor" aria-label="anchor" href="#samtools_idxstats_xy_plot"></a>
</h3>
<p>This table stores the X and Y chromosome expression information as
extracted by samtools. Note that I have found that looking at the top
500 most variable gene’s PCs is an easier method of determining sex (and
mislabels determined by sex) than these counts are. However, I use the
counts to confirm the PCA results.</p>
<table class="table">
<colgroup>
<col width="17%">
<col width="23%">
<col width="25%">
<col width="33%">
</colgroup>
<thead><tr class="header">
<th>Field</th>
<th>Purpose</th>
<th>Datatype</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>pk</td>
<td>primary key</td>
<td>INTEGER</td>
<td>autoincremented primary key</td>
</tr>
<tr class="even">
<td>sample_id</td>
<td>foreign key to sample</td>
<td>TEXT</td>
<td>References the “pk” field in the “sample” table</td>
</tr>
</tbody>
</table>
<details><summary>
Full Table Details
</summary><table class="table">
<colgroup>
<col width="17%">
<col width="23%">
<col width="25%">
<col width="33%">
</colgroup>
<thead><tr class="header">
<th>Field</th>
<th>Purpose</th>
<th>Datatype</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>pk</td>
<td>primary key</td>
<td>INTEGER</td>
<td>autoincremented primary key</td>
</tr>
<tr class="even">
<td>sample_id</td>
<td>foreign key to sample</td>
<td>TEXT</td>
<td>References the “pk” field in the “sample” table</td>
</tr>
<tr class="odd">
<td>chromosome_x</td>
<td></td>
<td>INTEGER</td>
<td>raw counts of reads over the X chromosome</td>
</tr>
<tr class="even">
<td>chromosome_y</td>
<td></td>
<td>INTEGER</td>
<td>raw counts of reads over the Y chromosome</td>
</tr>
</tbody>
</table></details>
</div>
<div class="section level3">
<h3 id="visitcode_map">visitcode_map<a class="anchor" aria-label="anchor" href="#visitcode_map"></a>
</h3>
<p>This is a mapping from <code>visit</code>, which is used in the
<code>fastq filename</code> to <code>visitcode</code>.
(<code>id/subject</code>, <code>visitcode</code>) form the primary key
of the LLFS phenotype data, while <code>visit</code>, though used in the
fastq filename, is not defined. This is the mapping that is used to get
the <code>visitcode</code> from the <code>visit</code>:</p>
<ul>
<li>if visit == 1, then visitcode = 1</li>
<li>if visit == 2, then visitcode is either 3 or 4</li>
<li>if visit == 3, then visitcode is either 6 or 7</li>
</ul>
<p>No subject can have visitcode 3 and 4, or visitcode 6 and 7, so this
mapping is actually one-to-one.</p>
<table class="table">
<colgroup>
<col width="13%">
<col width="23%">
<col width="8%">
<col width="53%">
</colgroup>
<thead><tr class="header">
<th>Field</th>
<th>Purpose</th>
<th>Datatype</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>pk</td>
<td>primary key autoincrement</td>
<td>INTEGER</td>
<td>Unique identifier for each record in the visitcode_map table</td>
</tr>
<tr class="even">
<td>whatdatall_id</td>
<td>foreign key</td>
<td>TEXT</td>
<td>References the “id” field in the “whatdatall” table</td>
</tr>
</tbody>
</table>
<p><detail></detail></p>
<summary>
Full Table Details
</summary><table class="table">
<colgroup>
<col width="13%">
<col width="23%">
<col width="8%">
<col width="53%">
</colgroup>
<thead><tr class="header">
<th>Field</th>
<th>Purpose</th>
<th>Datatype</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>pk</td>
<td>primary key autoincrement</td>
<td>INTEGER</td>
<td>Unique identifier for each record in the visitcode_map table</td>
</tr>
<tr class="even">
<td>whatdatall_id</td>
<td>foreign key</td>
<td>TEXT</td>
<td>References the “id” field in the “whatdatall” table</td>
</tr>
<tr class="odd">
<td>visit</td>
<td></td>
<td>TEXT</td>
<td>Represents a visit, only allows values ‘1’, ‘2’, ‘3’</td>
</tr>
<tr class="even">
<td>visitcode</td>
<td></td>
<td>TEXT</td>
<td>Represents a visit code, only allows values from ‘1’ to ‘7’</td>
</tr>
</tbody>
</table>
<p></p>
</div>
<div class="section level3">
<h3 id="wgs_compare">wgs_compare<a class="anchor" aria-label="anchor" href="#wgs_compare"></a>
</h3>
<p>This table stores the results of comparing the RNA sample variants to
WGS sample variants</p>
<table class="table">
<colgroup>
<col width="17%">
<col width="23%">
<col width="25%">
<col width="33%">
</colgroup>
<thead><tr class="header">
<th>Field</th>
<th>Purpose</th>
<th>Datatype</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>pk</td>
<td>primary key</td>
<td>INTEGER</td>
<td>autoincremented primary key</td>
</tr>
<tr class="even">
<td>sample_id</td>
<td>foreign key to sample</td>
<td>INTEGER</td>
<td>References “pk” of sample table</td>
</tr>
<tr class="odd">
<td>dna_subject</td>
<td></td>
<td>INTEGER</td>
<td>References “subject” of whatdatall table</td>
</tr>
</tbody>
</table>
<details><summary>
Full Table Details
</summary><table class="table">
<colgroup>
<col width="17%">
<col width="23%">
<col width="25%">
<col width="33%">
</colgroup>
<thead><tr class="header">
<th>Field</th>
<th>Purpose</th>
<th>Datatype</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>pk</td>
<td>primary key</td>
<td>INTEGER</td>
<td>autoincremented primary key</td>
</tr>
<tr class="even">
<td>sample_id</td>
<td>foreign key to sample</td>
<td>INTEGER</td>
<td>References “pk” of sample table</td>
</tr>
<tr class="odd">
<td>dna_subject</td>
<td></td>
<td>INTEGER</td>
<td>References “subject” of whatdatall table</td>
</tr>
<tr class="even">
<td>chr</td>
<td></td>
<td>TEXT</td>
<td>chromosome over which the comparison was conducted. Note that I
frequently use chr21 b/c it is shorter. I have done comparisons and
found that there is no loss of specificity by using a smaller
chromosome</td>
</tr>
<tr class="odd">
<td>total_variants</td>
<td></td>
<td>INTEGER</td>
<td>Total number of overlapping variants between the RNA and DNA</td>
</tr>
<tr class="even">
<td>matching_variants</td>
<td></td>
<td>INTEGER</td>
<td>Total number of overlapping variants which matched</td>
</tr>
<tr class="odd">
<td>homo_expr_cand</td>
<td></td>
<td>INTEGER</td>
<td>the total number of cases where the WGS is heterozygous but
expression homozygous</td>
</tr>
<tr class="even">
<td>match_ratio</td>
<td></td>
<td>REAL</td>
<td>matching_variants/total_variants</td>
</tr>
</tbody>
</table></details>
</div>
<div class="section level3">
<h3 id="whatdatall">whatdatall<a class="anchor" aria-label="anchor" href="#whatdatall"></a>
</h3>
<p>This table stores the <code>ID</code>, <code>subject</code>, and
<code>sex</code> as extracted from the LLFS phenotype data. Note that
this had to be augmented to include visit 3 samples, which are not yet
included in the publically shared LLFS <code>whatdatall</code>.</p>
<table class="table">
<thead><tr class="header">
<th>Field</th>
<th>Purpose</th>
<th>Datatype</th>
<th>Description</th>
</tr></thead>
<tbody><tr class="odd">
<td>pk</td>
<td>primary key</td>
<td>INTEGER</td>
<td>autoincremented primary key</td>
</tr></tbody>
</table>
<details><summary>
Full Table Details
</summary><table class="table">
<colgroup>
<col width="17%">
<col width="23%">
<col width="25%">
<col width="33%">
</colgroup>
<thead><tr class="header">
<th>Field</th>
<th>Purpose</th>
<th>Datatype</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>pk</td>
<td>primary key</td>
<td>INTEGER</td>
<td>autoincremented primary key</td>
</tr>
<tr class="even">
<td>id</td>
<td></td>
<td>INTEGER</td>
<td>the LLFS ID. Note that this is what is used in the fastq
filenames</td>
</tr>
<tr class="odd">
<td>subject</td>
<td></td>
<td>INTEGER</td>
<td>The LLFS subject identifier</td>
</tr>
<tr class="even">
<td>sex</td>
<td></td>
<td>INTEGER</td>
<td>labelled sex of the subject with this identifier</td>
</tr>
<tr class="odd">
<td>source</td>
<td></td>
<td>TEXT</td>
<td>where this record comes from</td>
</tr>
<tr class="even">
<td>notes</td>
<td></td>
<td>TEXT</td>
<td>any notes on this particular record</td>
</tr>
</tbody>
</table></details>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Chase Mateusiak.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
