<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>llfsRnaseq • llfsRnaseq</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="llfsRnaseq">
<meta property="og:description" content="llfsRnaseq">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">llfsRnaseq</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/llfsRnaseq.html">Get started</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Vignettes
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="dropdown-header">The RNAseq Database</li>
    <li>
      <a href="../articles/database-overview.html">Overview</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Processing Raw Data</li>
    <li>
      <a href="../articles/create-samplesheet.html">Running the nf-core/rnaseq pipeline</a>
    </li>
    <li>
      <a href="../articles/rna-dna-variant-compare.html">Running the modified nf-core/rnavar pipeline</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Initial QC</li>
    <li>
      <a href="../articles/sex-label-check.html">Sex Label Check</a>
    </li>
    <li>
      <a href="../articles/expression-qc.html">Expression QC</a>
    </li>
    <li>
      <a href="../articles/batch-effects.html">Batch Effects</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Creating a release</li>
    <li>
      <a href="../articles/creating-a-release.html">Setting filters and Exporting the Data</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">API</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
<li>
  <a href="https://nf-co.re/rnaseq/3.3/" class="external-link">nf-core/rnaseq pipeline</a>
</li>
<li>
  <a href="https://github.com/cmatKhan/rnavar" class="external-link">modified nf-core/rnavar</a>
</li>
<li>
  <a href="https://longlifefamilystudy.wustl.edu" class="external-link"></a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/cmatKhan/llfsRnaseq">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>llfsRnaseq</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/cmatKhan/llfsRnaseq/blob/HEAD/vignettes/llfsRnaseq.Rmd"><code>vignettes/llfsRnaseq.Rmd</code></a></small>
      <div class="hidden name"><code>llfsRnaseq.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>This is an R package which provides access to the code which I use to
process the LLFS RNAseq data.</p>
<p>Release data will now have a version, eg <code>1.0.0</code> which
refers to the version of this software. When a new release is made, the
<strong>major</strong> version number will increment. For example, if
the current version is <code>1.0.0</code>, then the release data will be
called <code>share_llfs_data_v1.0.0</code>. On the next release, I will
increase the version to <code>v2.0.0</code>, make a new software
release, and the shared llfs data will be
<code>share_llfs_data_v2.0.0</code>.</p>
<p>If there are any corrections to the code <strong>which affect the
data</strong>, but no new data is released, then I will increment the
<em>minor</em> version number. For example, <code>v1.1.0</code> would
mean that there is no new data from <code>v1.0.0</code>, but there was a
correction to code which affects the released data. There would then be
a data release called <code>share_llfs_data_v1.1.0</code>.</p>
<p>Finally, I will increment the <em>patch</em> version number if I make
a change to the code which does not affect the data. For example if I
just clean up some code in the vignettes or clarify some documentation.
This would look like <code>v1.0.1</code>, and there will be no new data
release associated with the change in the change in the <em>patch</em>
number.</p>
<p>Any change in the software will be described in detail in the <a href="https://cmatkhan.github.io/llfsRnaseq/news/index.html" class="external-link">CHANGELOG</a></p>
<p>** No LLFS Data, or anything which could be used to generate LLFS
Data or IDs, are shared in this repository. This is code only **</p>
<p>To get the data, talk to your lab and contact your points of contact
for the LLFS data administration. You do not need the code to use the
data, but if you are curious about how the data was processed from fastq
to release, then it is here.</p>
<p>Briefly, the steps are:</p>
<ol style="list-style-type: decimal">
<li>Receive the data from MGI</li>
<li>Use the MGI Demux sheet to create a input csv for the nf-core/rnaseq
pipeline
<ul>
<li>All of our data is processed with nf-core/rnaseq version 3.3</li>
</ul>
</li>
<li>Update the database with the new samples</li>
<li>Call variants on the RNAseq fastq files, convert the VCF to GDS
files, and then compare variants to the WGS data, if it exists for that
subject.</li>
<li>Update the database with the QC results from the RNAseq pipeline,
and the RNA vs WGS variants</li>
<li>Take a look at some feature so of the data (see articles/Initial
QC)</li>
<li>Update the NEWS.md file in this package, ensure that it builds
correctly, create a versioned release, and distribute the
share_lffs_data via box.</li>
</ol>
</div>
<div class="section level2">
<h2 id="the-database">The database<a class="anchor" aria-label="anchor" href="#the-database"></a>
</h2>
<p>The <a href="https://cmatkhan.github.io/llfsRnaseq/articles/database-overview.html" class="external-link">llfs_rnaseq_database.sqlite</a>
is now distributed along with the released LLFS RNAseq data. It is
intended to be a formally normalized database that allows me (and you,
if you want to use it) to re-name samples without losing any information
about what the original label was, or lose any connection to the metrics
which were generated under the original label. This database also serves
as a single collection of information related to these samples, such as
the “phantom sample” control sample information. I suggest interacting
with the database using any one (or more) of the following:</p>
<ul>
<li>
<strong>GUI</strong>: <a href="https://sqlitebrowser.org/" class="external-link">DB
Browser for SQLite</a>
</li>
<li>
<strong>R</strong>: <a href="https://rsqlite.r-dbi.org/" class="external-link">DBI</a>
</li>
<li>
<strong>Python</strong>: <a href="https://docs.python.org/3/library/sqlite3.html" class="external-link">sqlite3</a>
</li>
</ul>
</div>
<div class="section level2">
<h2 id="package-usage">Package Usage<a class="anchor" aria-label="anchor" href="#package-usage"></a>
</h2>
<div class="section level3">
<h3 id="requirements">Requirements<a class="anchor" aria-label="anchor" href="#requirements"></a>
</h3>
<ol style="list-style-type: decimal">
<li>R &gt;= 4.3.0</li>
</ol>
</div>
<div class="section level3">
<h3 id="how-to">How-to<a class="anchor" aria-label="anchor" href="#how-to"></a>
</h3>
<p>You may simply be interested in how this data is processed. In which
case, looking through the Articles (see the navbar) might be of
interest. You may want to process data yourself, or wish to run my code.
In that case, you should install the package onto your machine. I
suggest creating an R project to do this, but it is not necessary.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># note, decrease the number of CPUs as appropriate for your system</span></span>
<span><span class="co"># this is the official, or as official as anything in R gets, R development </span></span>
<span><span class="co"># toolset. It installs a lot of dependencies, and can take some time.</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">'usethis'</span>, Ncpus<span class="op">=</span><span class="fl">10</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># you can name this whatever you like. If you enter it like this, R will create </span></span>
<span><span class="co"># a project directory in your current working directory. If you would like to </span></span>
<span><span class="co"># put the project directory somewhere else, provide a relative or absolute path</span></span>
<span><span class="fu">usethis</span><span class="fu">::</span><span class="fu">create_project</span><span class="op">(</span><span class="st">'llfs_rnaseq_dataprocessing'</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># at this point, R will launch a new session which is in the package. The only </span></span>
<span><span class="co"># thing this means is that it is a dedicated directory with a .Rproj file. It </span></span>
<span><span class="co"># is more or less the same as just making a directory and putting all the stuff </span></span>
<span><span class="co"># related to whatever the 'project' is there.</span></span>
<span></span>
<span><span class="co"># this installs the llfsRnaseq R package</span></span>
<span><span class="fu">remotes</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">'https://github.com/cmatKhan/llfsRnaseq'</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="what-now">What now?<a class="anchor" aria-label="anchor" href="#what-now"></a>
</h3>
<div class="section level4">
<h4 id="using-the-package-as-a-user">Using the package as a user<a class="anchor" aria-label="anchor" href="#using-the-package-as-a-user"></a>
</h4>
<p>One thing you might want to do is open one of the vignettes. You can
see the available vignettes, after installing, with:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/vignette.html" class="external-link">vignette</a></span><span class="op">(</span>package<span class="op">=</span><span class="st">'llfsRnaseq'</span><span class="op">)</span></span></code></pre></div>
<p>You can pick and choose which you’re interested and copy/paste the
code into your own script or notebook. You can also visit the github
page and copy/paste the entire notebook if you wish (they are in the
vignettes/ directory). You will also have access to the functions
described in the Documentation’s API section.</p>
</div>
<div class="section level4">
<h4 id="using-the-package-as-a-developer">Using the package as a developer<a class="anchor" aria-label="anchor" href="#using-the-package-as-a-developer"></a>
</h4>
<p>In this case, just <code>git clone</code> the repo. The difference
between a “package” and a “project” is minimal – a “package” more or
less just has a <code>DESCRIPTION</code> file which allows the
<code>remotes::install_github()</code> command above to work. You can
treat this like a project directory and run the the notebooks in the
vignette, and <code>source</code> the code in the <code>R</code>
directory.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Chase Mateusiak.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
