<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>rna-dna-variant-compare • llfsRnaseq</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="rna-dna-variant-compare">
<meta property="og:description" content="llfsRnaseq">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">llfsRnaseq</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/llfsRnaseq.html">Get started</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Vignettes
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="dropdown-header">The RNAseq Database</li>
    <li>
      <a href="../articles/database-overview.html">Overview</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Processing Raw Data</li>
    <li>
      <a href="../articles/create-samplesheet.html">Running the nf-core/rnaseq pipeline</a>
    </li>
    <li>
      <a href="../articles/rna-dna-variant-compare.html">Running the modified nf-core/rnavar pipeline</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Initial QC</li>
    <li>
      <a href="../articles/sex-label-check.html">Sex Label Check</a>
    </li>
    <li>
      <a href="../articles/expression-qc.html">Expression QC</a>
    </li>
    <li>
      <a href="../articles/batch-effects.html">Batch Effects</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Creating a release</li>
    <li>
      <a href="../articles/creating-a-release.html">Setting filters and Exporting the Data</a>
    </li>
    <li>
      <a href="../articles/shared-data-overview.html">Shared Data Overview</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">API</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
<li>
  <a href="https://nf-co.re/rnaseq/3.3/" class="external-link">nf-core/rnaseq pipeline</a>
</li>
<li>
  <a href="https://github.com/cmatKhan/rnavar" class="external-link">modified nf-core/rnavar</a>
</li>
<li>
  <a href="https://longlifefamilystudy.wustl.edu" class="external-link"></a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/cmatKhan/llfsRnaseq">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>rna-dna-variant-compare</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/cmatKhan/llfsRnaseq/blob/HEAD/vignettes/rna-dna-variant-compare.Rmd"><code>vignettes/rna-dna-variant-compare.Rmd</code></a></small>
      <div class="hidden name"><code>rna-dna-variant-compare.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>The goal of this exercise is to call variants on the RNAseq samples.
These are the papers/resources I used to get an idea of what the state
of the art of calling variants from bulk RNAseq:</p>
<ol style="list-style-type: decimal">
<li><a href="https://gatk.broadinstitute.org/hc/en-us/articles/360035531192-RNAseq-short-variant-discovery-SNPs-Indels-" class="external-link">GATK
Calling Variants from Bulk RNAseq</a></li>
<li><a href="https://bcbio-nextgen.readthedocs.io/en/latest/contents/rnaseq_variants.html" class="external-link">bcbio
bulk RNAseq variant calling pipeline</a></li>
<li><a href="https://expert.cheekyscientist.com/how-to-do-variant-calling-from-rnaseq-ngs-data/" class="external-link">This
personal blog which includes some variant level filtering
suggestions</a></li>
</ol>
<p>The following are good papers on what to do with RNA variants. This
is offered as hopefully a spur to pursue these methods:</p>
<ol style="list-style-type: decimal">
<li>
<a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-015-0762-6" class="external-link">Tools
and best practices for data processing in allelic expression
analysis</a>
<ul>
<li>This is from 2015, so it is worth checking against more recent
papers. However, it seems still relevant, in particular the discussion
on WASP and the alignment step in general.</li>
</ul>
</li>
<li>
<a href="https://www.science.org/doi/10.1126/science.aaz5900" class="external-link">Transcriptomic
signatures across human tissues identify functional rare genetic
variation</a>
<ul>
<li>That’s gold Jerry! Gold!</li>
</ul>
</li>
<li><a href="https://academic.oup.com/bioinformatics/article/31/15/2497/187677?login=false" class="external-link">Assessing
allele-specific expression across multiple tissues from RNA-seq read
data</a></li>
<li><a href="https://pubmed.ncbi.nlm.nih.gov/31834882/" class="external-link">Leveraging
allelic imbalance to refine fine-mapping for eQTL studies</a></li>
</ol>
<p>Initially, the goal of the variant calling on the LLFS data is to
find and correct sample mislabels. The resources above, and my own trial
and error experimentation, show that it is indeed entirely reasonable to
call variants from the RNAseq samples and, where the same (labelled)
subject has WGS data, compare the RNA vs DNA variants. It turns out that
doing this on only one chromosome – any autosome will do, and I
frequently use chr21 – returns more than enough variant sites such that
there is no loss of specificity compared to comparing variants across
the entire genome.</p>
</div>
<div class="section level2">
<h2 id="procedure">Procedure<a class="anchor" aria-label="anchor" href="#procedure"></a>
</h2>
<p>I use a modified version of the nf-core/rnavar RNA variant calling
pipeline, which is based on the GATK protocol and uses their tools. You
can find my modifications <a href="https://github.com/cmatKhan/rnavar/tree/bam_input" class="external-link">here</a>. This
allows for <strong>bam</strong> input, rather than fastq, and allows us
to use the bams output by the RNAseq pipeline as opposed to starting
from the alignment step. I process these in batches of no more than 30 –
due to the way nextflow handles tracking intermediate data, and the
number of times the bam files have to be manipulated, and therefore
copied, the size of the <code>work</code> directory gets very large,
upwards of 10 terabytes.</p>
<ol style="list-style-type: decimal">
<li><p>Create an input samplesheet. An example of the <code>bam</code>
input samplesheet can be found <a href="https://github.com/cmatKhan/llfsRnaseq/blob/main/inst/20230519.csv">here</a></p></li>
<li>
<p>Run the pipeline.</p>
<ul>
<li>First, <a href="">clone the repo</a> and make sure that you switch
to the <code>bam_input</code> branch.</li>
<li>
<a href="https://github.com/cmatKhan/llfsRnaseq/blob/main/inst/run_nf.sh">This
is an example</a> of the script I use to run the pipeline on HTCF (a
cluster which uses SLURM). I have included <a href="https://github.com/cmatKhan/llfsRnaseq/blob/main/inst/htcf.config">the
config file</a> for HTCF, also.</li>
</ul>
</li>
<li><p>I have the pipeline to set to symlink rather than copy files from
<code>work</code> to <code>results</code>, so when you move the
<code>vcf</code> and <code>tabix</code> files from <code>scratch</code>
to whatever long term storage you’re using (brentlab HTCF users, I use
LTOS), make sure you tell the copy program to follow the symlinks and
grab the actual file, for instance <code>--copy-links</code> using
<code>rsync</code> or <code>-F</code> using <code>s3cmd</code>.</p></li>
<li><p>Unless you’re doing 30 or less samples, you’ll need to repeat
this procedure in batches of ~30.</p></li>
<li><p>At this point, I use<br><a href="https://github.com/cmatKhan/llfs_rnaseq_variants" class="external-link">the scripts
in this R project</a> to convert the <code>vcf</code> files to
<code>gds</code> files, and then use the WGS dna <code>gds</code> files
to extract and compare variants (also with scripts in that
repo).</p></li>
<li><p>Finally, I add the result of at least the direct comparison to
the <code>llfs_rnaseq_database</code>. I haven’t yet made a table to
track the summarized results of the all-by-all comparisons, where they
are necessary for the obvious mislabels.</p></li>
</ol>
<p>A work in progress is taking all of the gvcf files (the modified
nf-core/rnavar pipeline outputs put vcf and gvcf files), combining them,
and then using hte GATK joint variant caller. Doing this would allow the
calculation of a kinship matrix, which may help further confirm the
re-labelling, and also possibly identify and hopefully provide evidence
for re-labelling mislabeled samples which do not have WGS data.</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Chase Mateusiak.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
